core/src/main/java/hudson/PluginManager.java

```diff
@@ -1182,22 +1198,30 @@ private void trackInitialPluginInstall(@Nonnull final List<Future<UpdateCenter.U
             new Thread() {
                 @Override
                 public void run() {
+                    final boolean failures[] = { false };
```

Comment:
IMHO this trick is only needed when you need to mutate variable declared outside of anonymous class from inside of its method. (Such reference needs to be final so to make it modifiable it can be wrapped in final array (with mutable elements))